src/Microsoft.ML.StaticPipe/StaticSchemaShape.cs

```diff
@@ -327,34 +327,21 @@ private static Type GetTypeOrNull(Schema.Column col)
         /// type for communicating text.
         /// </summary>
         /// <returns>The basic type used to represent an item type in the static pipeline</returns>
-        private static Type StaticKind(DataKind kind)
+        private static Type GetPhysicalType(ColumnType columnType)
         {
-            switch (kind)
+            switch (columnType)
             {
-                // The default kind is reserved for unknown types.
-                case default(DataKind): return null;
-                case DataKind.I1: return typeof(sbyte);
-                case DataKind.I2: return typeof(short);
-                case DataKind.I4: return typeof(int);
-                case DataKind.I8: return typeof(long);
-
-                case DataKind.U1: return typeof(byte);
-                case DataKind.U2: return typeof(ushort);
-                case DataKind.U4: return typeof(uint);
-                case DataKind.U8: return typeof(ulong);
-                case DataKind.U16: return typeof(RowId);
-
-                case DataKind.R4: return typeof(float);
-                case DataKind.R8: return typeof(double);
-                case DataKind.BL: return typeof(bool);
-
-                case DataKind.Text: return typeof(string);
-                case DataKind.TimeSpan: return typeof(TimeSpan);
-                case DataKind.DateTime: return typeof(DateTime);
-                case DataKind.DateTimeZone: return typeof(DateTimeOffset);
+                case NumberType numberType:
+                case TimeSpanType timeSpanType:
+                case DateTimeType dateTimeType:
+                case DateTimeOffsetType dateTimeOffsetType:
+                case BoolType boolType:
+                    return columnType.RawType;
+                case TextType textType:
+                    return typeof(string);
 
                 default:
-                    throw Contracts.ExceptParam(nameof(kind), $"Unrecognized type '{kind}'");
+                    return null;
```

Comment:
>return null; [](start = 20, length = 12)

curious, why change to return null, rather than throw? If an invalid type gets passed, would it be better to throw?

Reply:
I was trying to preserve the first line's behavior:

```
case default(DataKind): return null;
```

When DataKind is `default`, it usually means it is a "custom" or "structured" type - like `VectorType` or `ImageType`. So I was preserving that behavior here - when you see an unrecognized `ColumnType` - return `null`.

I assume the existing `throw` condition was in place for handling invalid `enum` values - like `29` and `-451`.

---
In reply to: [247782795](https://github.com/dotnet/machinelearning/pull/2143#discussion_r247782795) [](ancestors = 247782795)